internal-webapps/org.wso2.bfsi.consent.management.endpoint/src/main/java/org/wso2/bfsi/consent/management/endpoint/utils/ConsentUtils.java

```diff
@@ -0,0 +1,508 @@
+                BasicClientCookie cookie = new BasicClientCookie(cookieValue.getKey(), cookieValue.getValue());
+                cookie.setDomain(cookieDomain);
+                cookie.setPath("/");
+                cookie.setSecure(true);
+                cookieStore.addCookie(cookie);
+            }
+            HttpPost authorizeRequest = new HttpPost(authorizeURL);
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.HAS_APPROVED_ALWAYS, "false"));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.SESSION_DATA_KEY_CONSENT,
+                    consentData.getSessionDataKey()));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.CONSENT, consent));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.USER, consentData.getUserId()));
+            HttpContext localContext = new BasicHttpContext();
+            localContext.setAttribute(HttpClientContext.COOKIE_STORE, cookieStore);
+            UrlEncodedFormEntity entity = new UrlEncodedFormEntity(params);
+            authorizeRequest.setEntity(entity);
+            HttpResponse authorizeResponse = client.execute(authorizeRequest, localContext);
+
+            if (authorizeResponse.getStatusLine().getStatusCode() != HttpURLConnection.HTTP_MOVED_TEMP) {
+                throw new ConsentException(consentData.getRedirectURI(), AuthErrorCode.SERVER_ERROR,
+                        "Error while getting authorize redirect", consentData.getState());
+            } else {
+                //Extract the location header from the authorization redirect
+                return new URI(authorizeResponse.getLastHeader(ConsentExtensionConstants.LOCATION).getValue());
+            }
+        } catch (IOException e) {
+            log.error("Error while sending authorize request to complete the authorize flow", e);
+            return null;
+        } catch (URISyntaxException e) {
+            log.error("Authorize response URI syntax error", e);
+            throw new ConsentException(consentData.getRedirectURI(), AuthErrorCode.SERVER_ERROR,
+                    "Internal server error", consentData.getState());
+        }
+    }
+
+    public static JSONObject detailedConsentToJSON(DetailedConsentResource detailedConsentResource) {
+        JSONObject consentResource = new JSONObject();
+
+        consentResource.put(ConsentExtensionConstants.CC_CONSENT_ID, detailedConsentResource.getConsentID());
+        consentResource.put(ConsentExtensionConstants.CLIENT_ID, detailedConsentResource.getClientID());
+        try {
+            consentResource.put(ConsentExtensionConstants.RECEIPT,
+                   new JSONObject(detailedConsentResource.getReceipt()));
+        } catch (JSONException e) {
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Exception occurred while parsing" +
```

Comment:
Some places are found without the throwable object. Shall we add them?

Reply:
We are logging the exception and return a meaningful error. Even though we pass the exception it will not reach the end customer